### DISCUSSION TITLE
Get latest succeeded commit hash after deployment failures (Bitbucket)

### DISCUSSION THREAD

**Discussion by sjurgis** [scolladon/sfdx-git-delta#595] [General]
Apologies, but this is more of a practical question for community members, than about sgd.

Lets say deployment #1 fails and a fix is submitted. Comparing changes between HEAD~1 and HEAD will discard changes from deployment #1.

My idea is to query last n pipeline runs to get commit hash from latest successful run and then compare that to HEAD.

How do you solve this? Is there some gold standard?

**Comment by scolladon** [comment]
Hi @sjurgis 

I think there are multiple ways to deal with this kind of situation.
Depending on your branching strategy it can be more or less straightforward.

What I can suggest here is to use git tags with a format you can easily parse and compare.
Per exemple you could tag every deployment done with a format like `dd-mm-yyyy-env` (For example: 25-11-2022-sit)
Then when deploying you could compare the actual state of the repo with the state of the repo related to the latest tag

Or you could have only one tag per env. Each time deployment is done on the env you could move the tag.
I don't really like this solution because moving tag is not great in CI and also it can be confusing for others.

I'm sure there are many other workflows, and I can't wait to see those 😄

**Comment by AllanOricil** [comment]
I'm using tags with this format


`state/metadata/ENV_NAME`


when a deploy succeeds in a particular environment I update this tag in remote with the hash that deployed the changes.


I also have tags for vlocity datapacks

`state/datapacks/ENV_NAME`


Approved changes that can go to production are tagged with `YYYY-MM-DD` (I don't use semver because there are multiple apps from different domains and we did not separate things in packages)

**Reply by sjurgis** [reply]
Thanks - I've implemented something similar.
